stan/math/prim/mat/fun/gp_exp_quad_cov.hpp

```diff
@@ -2,17 +2,25 @@
 #define STAN_MATH_PRIM_MAT_FUN_GP_EXP_QUAD_COV_HPP
 
 #include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/arr/meta/get.hpp>
```

Comment:
I think there's a bit of confusion here. A lot of these headers shouldn't be included. The includes should look like:
```
#include <stan/math/opencl/err/check_nan.hpp>
#include <stan/math/opencl/gp_exp_quad_cov.hpp>
#include <stan/math/prim/mat/fun/Eigen.hpp>
#include <stan/math/prim/mat/fun/divide_columns.hpp>
#include <stan/math/prim/mat/fun/squared_distance.hpp>
#include <stan/math/prim/scal/err/check_not_nan.hpp>
#include <stan/math/prim/scal/err/check_positive.hpp>
#include <stan/math/prim/scal/err/check_positive_finite.hpp>
#include <stan/math/prim/scal/err/check_size_match.hpp>
#include <stan/math/prim/scal/fun/exp.hpp>
#include <stan/math/prim/scal/fun/square.hpp>
#include <stan/math/prim/scal/meta/return_type.hpp>
#include <cmath>
#include <type_traits>
#include <vector>
```

(All the Math stuff in alphabetical order, then all the standard library stuff afterwards.)

I just ran these two tests and it's fine. Where do you see an error?
```
./runTests.py test/unit/math/prim/mat/fun/gp_exp_quad_cov_test.cpp
./runTests.py test/unit/math/rev/mat/fun/gp_exp_quad_cov_test.cpp
```

Reply:
So here's the weird, these also pass locally for me but fail on Jenkins. It looks like Jenkins is using clang++6, but my computer is having a hard time compiling gtest with that compiler

Reply:
I'll check the output on Jenkins to see if I spot something.

Reply:
> I just ran these two tests and it's fine. Where do you see an error?

So the errors actually happen in some pretty weird places that are not the gp_exp_quad_cov test. But the order of headers in `prim/mat/fun/gp_exp_quad_cov.hpp` is what's goofing them up

For instance in the below we have an error in `test/unit/math/rev/core/ddv_vari_test`

http://d1m1s1b1.stat.columbia.edu:8080/blue/organizations/jenkins/Math%20Pipeline/detail/PR-1206/15/pipeline/#step-107-log-1393

make: *** [test/unit/math/rev/core/ddv_vari_test.o] Error 1

Reply:
Also I just pushed up the changes to the header order you mentioned above

Reply:
So... it runs on my machine locally also. That said, the test might be faulty. There really should be a `recover_memory()` between tests, although this never flexes reverse mode, so not sure what it's really testing.

The thing I'm worried about is that it's triggering some unsafe memory condition. Maybe when it's allocating new memory? But I'm not super-worried about it now if it passes. (It could have also just run out of memory.)